# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.cpp
# Commit: f946921b9173
# Full Hash: f946921b9173529d6ceec8200469747446c98a6a
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2022-05-23 09:52:44
# Regressor Bug: 1770539
# File Overlap Count: 1
# Description:
#   Bug 1770539 p3 - Move nsPrintData::mShrinkRatio to nsPrintJob::mShrinkToFitFactor. r=emilio
#   
#   Depends on D146974
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146975
# ==============================================================================

diff -r 5cd20e839a73 -r f946921b9173 layout/printing/nsPrintJob.cpp
--- a/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:56 2022 +0000
+++ b/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:56 2022 +0000
@@ -832,9 +832,9 @@
   // Here is where we figure out if extra reflow for shrinking the content
   // is required.
   if (mShrinkToFit) {
-    printData->mShrinkRatio = printData->mPrintObject->mShrinkRatio;
+    mShrinkToFitFactor = printData->mPrintObject->mShrinkRatio;
 
-    if (printData->mShrinkRatio < 0.998f) {
+    if (mShrinkToFitFactor < 0.998f) {
       nsresult rv = ReconstructAndReflow();
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
@@ -853,8 +853,7 @@
           ("*******************************************************************"
            "*******\n"));
       PR_PL(("STF Ratio is: %8.5f Effective Ratio: %8.5f Diff: %8.5f\n",
-             printData->mShrinkRatio, calcRatio,
-             printData->mShrinkRatio - calcRatio));
+             mShrinkToFitFactor, calcRatio, mShrinkToFitFactor - calcRatio));
       PR_PL(
           ("*******************************************************************"
            "*******\n"));
@@ -1137,7 +1136,7 @@
 void nsPrintJob::UpdateZoomRatio(nsPrintObject* aPO) {
   if (!aPO->mParent) {
     if (mShrinkToFit) {
-      aPO->mZoomRatio = mPrt->mShrinkRatio;
+      aPO->mZoomRatio = mShrinkToFitFactor;
       // If we're actually going to scale (the factor is less than 1), we
       // reduce the scale factor slightly to avoid the possibility of floating
       // point rounding error causing slight clipping of the longest lines.